### PR TITLE
Sf3 compatibility

### PR DESCRIPTION
**Pull request by greg0ire**
I am targetting this branch, because this is BC.

Closes #249 

## Changelog

```markdown
### Fixed
- Symfony 3 compatibility was improved
```

## Subject

This PR fixes issues raised in #249 , but in a backwards-compatible manner.

### CODE REVIEW

**Comment by greg0ire** [comment]
@nicodmf, please review and test this.

File: Admin/CategoryAdmin.php [hunk 1]
```diff
@@ -18,10 +18,6 @@
 
 class CategoryAdmin extends ContextAwareAdmin
 {
-    protected $formOptions = array(
```
Comment (core23) [inline Admin/CategoryAdmin.php]: Why did you remove this?

File: Resources/config/validation.xml [hunk 2]
```diff
@@ -9,6 +9,15 @@
...
         </property>
+        <property name="parent">
+            <constraint name="Valid"/>
+        </property>
+        <property name="media">
+            <constraint name="Valid"/>
+        </property>
+        <property name="context">
+            <constraint name="Valid"/>
+        </property>
```
Comment (greg0ire) [inline Resources/config/validation.xml]: @core23 : this is what replaces `cascade_validation`

File: Admin/CategoryAdmin.php (hunk 1 above)
Comment (greg0ire) [inline Admin/CategoryAdmin.php]: More details about this in [the commit message](https://github.com/sonata-project/SonataClassificationBundle/pull/250/commits/1d98b7549b420bf53faabd1871cde38aed26e47e)

Comment (greg0ire) [inline Admin/CategoryAdmin.php]: @core23 what changes are you requesting exactly?

**Comment by greg0ire** [comment]
@Grazulex @gremo @vub please review and test.

**Comment by greg0ire** [comment]
@gremo , I did what you suggested, please test.

File: Admin/CategoryAdmin.php [hunk 3]
```diff
@@ -33,6 +35,26 @@ public function configureRoutes(RouteCollection $routes)
...
+        if (isset($this->formOptions['cascade_validation'])) {
+            unset($this->formOptions['cascade_validation']);
+            $this->formOptions['constraints'][] = new Valid();
+        } else {
+            @trigger_error(<<<EOT
+Unsetting cascade_validation is deprecated since 3.x, and will give an error in 4.0.
+Override getFormBuilder() and remove the "Valid" constraint instead.
+EOT
+            , E_USER_DEPRECATED);
+        }
```
Comment (greg0ire) [inline Admin/CategoryAdmin.php]: @gremo : seems correct?

Comment (greg0ire) [inline Admin/CategoryAdmin.php]: ping @gremo

Comment (greg0ire) [inline Admin/CategoryAdmin.php]: Yeah, probably not ideal indeed, but I can't think of another way either. Plus, I'm not sure many people would actually want to disable the validation. What do you think?

Comment (gremo) [inline Admin/CategoryAdmin.php]: LGTM. Really thank you for helping, can I ask you when a new stable will be released? I don't know the bundle release cycle yet.

**Comment by greg0ire** [comment]
Thanks @core23 ! @sonata-project/contributors , final review and merge?

File: Form/Type/CategorySelectorType.php [hunk 4]
```diff
@@ -40,10 +41,17 @@ public function __construct(ManagerInterface $manager)
     }
 
     /**
+     * NEXT_MAJOR: Remove method, when bumping requirements to SF 2.7+.
+     *
      * {@inheritdoc}
      */
     public function setDefaultOptions(OptionsResolverInterface $resolver)
     {
+        $this->configureOptions($resolver);
```
Comment (OskarStark) [inline Form/Type/CategorySelectorType.php]: trigger_error ?

Comment (greg0ire) [inline Form/Type/CategorySelectorType.php]: We don't do it in that case for other bundles, at least not on the methods I found, maybe because the only consumer of this method is Symfony? Or maybe it already throws a deprecation? Or because we don't want people in 2.3 (for instance) to get a deprecation notice for that?

Comment (greg0ire) [inline Form/Type/CategorySelectorType.php]: ping @OskarStark

Comment (OskarStark) [inline Form/Type/CategorySelectorType.php]: ok fine for me!

**Comment by OskarStark** [comment]
Thank you @greg0ire 👍

**Comment by greg0ire** [comment]
Thanks @OskarStark !
@gremo : asking a release rightaway

**Comment by nicodmf** [comment]
@greg0ire Thanks a lot :)
